gnat2goto/driver/tree_walk.adb

```diff
@@ -3386,7 +3394,8 @@ package body Tree_Walk is
          return Ret;
       end if;
       Set_Lhs (Ret, LHS);
-      Set_Rhs (Ret, RHS);
+      Set_Rhs (Ret, Typecast_If_Necessary (RHS, Get_Type (LHS),
```

Comment:
when are the types of LHS and RHS different, and when they are why is it correct to typecast RHS to the type of LHS rather than the other way round?